lib_xua/src/core/mixer/mixer.xc

```diff
@@ -439,11 +442,52 @@ static void mixer1(chanend c_host, chanend c_mix_ctl, chanend ?c_mixer2, chanend
         /* Request from audio()/mixer2() */
         request = inuint(c_audio);
 
-        /* Forward on Request for data to decouple thread */
+        // Check if cmd is pending from last exchange with decoupler.
+        // If no command pending, transfer sample with audiohub to unblock it as
+        // soon as possible. This is esp important for the I2S TDM case where lots
+        // more samples need to be exchanged in lesser time than regular I2S
+        // (for example, for 96KHz tdm8. 4 i2s lines, exchange 32 samples in (1/(96000*8))seconds)
+        if(!cmd_pending_flag)
+        {
+            GiveSamplesToDevice(c_audio, samples_to_device_map);
+            GetSamplesFromDevice(c_audio);
+        }
+        else
+        {
+            // forward cmd from last exchange with decoupler to audio
+            switch(pending_cmd[0])
+            {
+                case XUA_AUDCTL_SET_SAMPLE_FREQ:
```

Comment:
Wondering if we should have a common comms structure for commands - always 3 words or simular

then we can pass things though without having to interrogate the actual command and remove this whole switch{}

Reply:
I've done the change in https://github.com/xmos/lib_xua/pull/535/commits/d5287d3e35474444381113eeb06f1dc3bad2b57b.
Ideally, in audiohub also, I'd like to read the command and data in one go, instead of first getting command, exiting audiohub main loop and then receiving the data, but that would required changing the API of a lot of audiohub functions which I don't want to do, since not sure of implication on timing.

Reply:
I tried doing this and it worked but required a lot more change than I anticipated ([commit saved in](https://github.com/xmos/lib_xua/commit/1b0027cb30c7435b39ad3fbdfccaba65ac3fc59e)), so have undone the change. I'll create an issue to do this properly later.

Reply:
https://github.com/xmos/lib_xua/issues/537